src/components/talks/AnimatedSlide/index.tsx

```diff
@@ -0,0 +1,154 @@
+              togglePlayState();
+              isFirstEntryRef.current = false;
+            }
+          });
+        },
+        { threshold },
+      );
+
+      observer.observe(svgRef.current.base);
+    }
+
+    const elements = svgRef.current.base?.querySelectorAll<SVGGElement>('g g');
+    runningAnimationsCountRef.current = elements?.length ?? 0;
+
+    const onAnimationEnd = () => {
+      runningAnimationsCountRef.current = runningAnimationsCountRef.current - 1;
+
+      if (runningAnimationsCountRef.current === 0) {
+        setIsFinished(true);
+      }
+    };
+
+    elements?.forEach((element) => {
+      element.addEventListener('animationend', onAnimationEnd);
+    });
+
+    return () => {
+      elements?.forEach((element) => {
+        element.removeEventListener('animationend', onAnimationEnd);
+      });
+
+      observer?.disconnect();
+    };
+  }, [isRestarting]);
+
+  const changeAnimationPlayState = (state: 'running' | 'paused') => {
+    svgRef.current?.base
+      ?.querySelectorAll<SVGGElement>('g g')
+      .forEach((element) => (element.style.animationPlayState = state));
+  };
+  const togglePlayState = () => {
+    setIsPaused((prevIsPaused) => {
+      changeAnimationPlayState(prevIsPaused ? 'running' : 'paused');
+      return !prevIsPaused;
+    });
+  };
+  useEffect(() => {
+    if (autoplay === 'enabled') {
+      changeAnimationPlayState('running');
+    }
+  }, []);
+
+  const restart = () => {
+    // super duper ugly hacks to restart svg animation!
+    setIsFinished(false);
+    setIsRestarting(true);
+    window.requestAnimationFrame(() => {
+      setIsRestarting(false);
+      window.requestAnimationFrame(() => changeAnimationPlayState('running'));
+    });
+  };
+
+  const componentContents = (
+    <Container className={className} id={slideId}>
+      <div>
+        <ImageWrapper
+          href={`#${slideId}`}
+          useImageBorder={useImageBorder}
+          isSectionHeader={isSectionHeader}
+        >
+          <Svg
+            style={{
+              height: 'auto',
+              display: 'block',
+            }}
+            viewBox="0 0 1280 720"
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            ref={svgRef}
+          />
+        </ImageWrapper>
+        {hasControls && !isFinished && (
+          <PlayControlButton onClick={togglePlayState}>
+            {isPaused ? 'Play' : 'Pause'}
+          </PlayControlButton>
+        )}
+        {hasControls && isFinished && (
+          <PlayControlButton onClick={restart}>Replay</PlayControlButton>
+        )}
+      </div>
+      <Text isSectionHeader={isSectionHeader}>{children}</Text>
+    </Container>
+  );
+
+  // why? to replace svg node with finished animations with newly created one, which would allow to restart animation.
+  // this turned out to be the only way to properly do it.
+  if (isRestarting) {
+    return <div>{componentContents}</div>;
+  }
```

Comment:
BTW a great trick to re-mount a component is to do this:

```js
<div key={someKey}>...</div>
```

and then change the key whenever you need to re-mount the component. (This probably isn’t that relevant here as we’ll replace this logic with an inline script, but might be useful in the future!)